apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java

```diff
@@ -745,28 +743,26 @@ protected String buildPathStartingFromBaseDir(String... pathTokens) {
    }
 
    /**
-    * Substitutes all the file separator occurrences in the path with a file separator for the
-    * current operative system
+    * Convert path to the current OS filesystem standard
     *
-    * @param pathToBeNormalized
+    * @param path
     * @return
     */
-   private static String normalize(String pathToBeNormalized) {
-      if (null != pathToBeNormalized && pathToBeNormalized.contains(BACK_SLASH)) {
-         if (!BACK_SLASH.equals(File.separator)) {
-            return pathToBeNormalized.replace(BACK_SLASH, File.separator);
-         }
+   private static String normalize(String path) {
+      if (null != path) {
```

Comment:
But then it is not cross-platform. I mean, it's fairly easy to put the exception in, but a real fix will probably be somewhat more complex (i.e. hex-encoding the blob name). But I do get we want to keep this a filesystem provider, so it makes sense to have it follow the filesystem closely.